arangod/Aql/Executor/EnumerateNearVectorExecutor.cpp

```diff
@@ -41,132 +49,197 @@ using Stats = NoStats;
+    }
+    fillOutput(output);
+    if (_currentProcessedResultCount == _infos.getNumberOfResults()) {
+      inputRange.advanceDataRow();
+      _initialized = false;
+      break;
+    }
+  }
 
-  AqlCall upstreamCall{};
-  upstreamCall.fullCount = output.getClientCall().fullCount;
+  return {inputRange.upstreamState(), {}, output.getClientCall()};
+}
 
-  std::vector<float> inputRowsJoined;
-  while (inputRange.hasDataRow() && !output.isFull() &&
-         _currentProcessedResultCount == _inputRows.size() * _infos.topK) {
-    _initialized = false;
-    fillInput(inputRange, inputRowsJoined);
-  }
-  if (_inputRows.empty()) {
-    return {inputRange.upstreamState(), stats, output.getClientCall()};
-  }
+[[nodiscard]] std::tuple<ExecutorState, Stats, size_t, AqlCall>
+EnumerateNearVectorsExecutor::skipRowsRange(AqlItemBlockInputRange& inputRange,
+                                            AqlCall& call) {
+  std::size_t skipped{0};
+  LOG_INTERNAL << "call: " << call << " fullCount: " << std::boolalpha
+               << call.fullCount << " needSkipMore: " << std::boolalpha
+               << call.needSkipMore()
+               << ", hasInputRange: " << inputRange.hasDataRow()
+               << ", resultsAreProcessed: " << _resultsAreProcessed;
+  while (call.needSkipMore() &&
+         (inputRange.hasDataRow() || !_resultsAreProcessed)) {
+    // get an input row first, if necessary
+    if (!_inputRow.isInitialized() && !_initialized) {
+      std::tie(_state, _inputRow) = inputRange.peekDataRow();
+
+      if (_inputRow.isInitialized()) {
+        fillInput(inputRange);
+        searchResults();
+      } else {
+        break;
+      }
+    }
 
-  if (!_initialized && !_inputRows.empty()) {
-    searchResults(inputRowsJoined);
-  }
+    LOG_INTERNAL << "offset: " << call.getOffset()
+                 << ", _currentProcessedResultCount: "
+                 << _currentProcessedResultCount;
+    if (call.getOffset() == 0) {
+      TRI_ASSERT(call.needsFullCount())
+          << "if we need to skip and skip is 0 fullCount must be true";
+      LOG_INTERNAL << "needsFullCount is set";
 
-  if (_inputRows.size() != 0) {
-    fillOutput(output);
+      skipped = ExecutionBlock::SkipAllSize();
+      break;
+    }
+
+    if (call.getOffset() == 0) {
+      TRI_ASSERT(call.needsFullCount());
+      // we are done
+      skipped = ExecutionBlock::SkipAllSize();
+      inputRange.advanceDataRow();
+      _inputRow = InputAqlItemRow{CreateInvalidInputRowHint{}};
+      _initialized = false;
+      continue;
+    }
+
+    while (call.getOffset() > 0) {
+      if (_labels[_currentProcessedResultCount] == -1) {
+        // we are done, since faiss return -1 for nonexistent results
+        skipped = ExecutionBlock::SkipAllSize();
+        _resultsAreProcessed = true;
+        inputRange.advanceDataRow();
+        _inputRow = InputAqlItemRow{CreateInvalidInputRowHint{}};
+        _initialized = false;
+        call.didSkip(skipped);
```

Comment:
No, it is not... I remember we spoke about this being the count that we can get from collection in EnumerateNearVectorNode, but the issue is I cannot access the collection count since I need transaction object here, and I believe that does not exists yet...

Reply:
Yes, you are right. Sorry forthe  confusion. I have managed to solve this, and added the coverage in tests